lib/katello/tasks/fix_hostgroup_facets.rake

```diff
@@ -0,0 +1,39 @@
+namespace :katello do
+  desc "This task collates hostgroup content facts that were missed during the upgrade from audit.\
+        It then updates the hostgroup content_facet accordingly."
+  task :fix_hostgroup_facets => :environment do
+    def interested_hostgroups
+      Hostgroup.unscoped.where(id: Katello::Hostgroup::ContentFacet.
+                          where(content_source_id: nil,
+                                kickstart_repository_id: nil,
+                                content_view_id: nil,
+                                lifecycle_environment_id: nil).select(:hostgroup_id))
+    end
+
+    def pick_facet_values(hg)
+      facet_values = {}
+      hg.audits.reverse_each do |audit|
```

Comment:
Yeh hg.audits automatically orders it by versions. I tried `hg.audits.order(version: :desc)` and that didnt work (since I think it has a default order.)